packages/rescript-relay-router/cli/RescriptRelayRouterCli__Utils.res

```diff
@@ -85,6 +85,8 @@ module QueryParams = {
     | Boolean => `switch ${variableName} { | true => "true" | false => "false" }`
     | Int => `Int.toString(${variableName})`
     | Float => `Float.toString(${variableName})`
+    | CustomModule({moduleName, required: true}) =>
+      `${variableName} == ${moduleName}.defaultValue ? None : ${variableName}->${moduleName}.serialize->Some`
```

Comment:
I don't have a better solution, but I don't think `==` works well enough to be used in this context. IIRC there are a few issues around newer representations we've implemented. I could be wrong though.

The only other way I can think of to do it is to serialize before comparing. That'll give a stable comparable value. It _could_ be hurtful to performance if serializing is expensive, but that's probably rarely the case, and it's not done often. At that point one could consider caching the serialization of the default value, or something along those lines.

What do you think about serializing before comparing?

Reply:
It'd make sense, anyway we'd serialize the value without this feature, so it's the same complexity!